File: exercism/cpp/cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
	unsigned square_of_sum(unsigned num) {
		unsigned sum = (num * (num + 1)) / 2;
		return sum * sum;
	}

	unsigned sum_of_squares(unsigned num) {
		unsigned res = 0;

		for (unsigned i = 1; i <= num; num++) {
			res += i * i;
		}
		return res;
	}

	unsigned difference(unsigned num) {
		return square_of_sum(num) - sum_of_squares(num);
	}

}  // namespace difference_of_squares

File: exercism/go/go/difference-of-squares/difference_of_squares.go
// Package diffsquares finds the difference between the squares of sums
// and sum of squares
package diffsquares

// SquareOfSum returns the square of summed values
func SquareOfSum(n int) int {
	var res int

	res = (n * (n + 1)) / 2

	return res * res
}

// SumOfSquares returns the sum of squares
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference returns the difference of SquareOfSum and SumOfSquares
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
